src/server/system_services/bucket_server.js

```diff
@@ -247,6 +251,49 @@ function validate_pool_constraints({ mongo_pool, default_pool }) {
     }
 }
 
+function validate_replication(req) {
+    const replication_policy = req.rpc_params.replication_policy;
+    // num of rules in configuration must be in defined limits
+    if (replication_policy.length > config.BUCKET_REPLICATION_MAX_RULES ||
+        replication_policy.length < config.BUCKET_REPLICATION_MIN_RULES) throw new RpcError('INVALID_REPLICATION_POLICY');
```

Comment:
< config.BUCKET_REPLICATION_MIN_RULES fails, so how do I remove rules altogether (getting the bucket to 0 rules)

Reply:
I decided to add delete_bucket_replication function for deletion of all rules. WDYT?

Reply:
So update for deletion of a single rule and delete replication for all?
It's weird that I can delete one by one besides the last one , WDYT? Maybe add a way to remove all rules with the same API of update ?

Reply:
I can remove the min for now and update will work as you want but then delete_bucket_replication is redundant and we won't be compatible with s3 bucket replication

Reply:
If its for compatibility, lets keep it